app/controllers/api/v4/call_results_controller.rb

```diff
@@ -0,0 +1,35 @@
+class Api::V4::CallResultsController < Api::V4::SyncController
+  def sync_from_user
+    __sync_from_user__(call_result_params)
+  end
+
+  private
+
+  def merge_if_valid(call_result_params)
+    validator = Api::V4::CallResultPayloadValidator.new(call_result_params)
+    logger.debug "CallResult had errors: #{validator.errors_hash}" if validator.invalid?
+    if validator.check_invalid?
+      {errors_hash: validator.errors_hash}
+    else
+      transformed_params = Api::V4::Transformer.from_request(call_result_params)
+      call_result = CallResult.merge(transformed_params)
+
+      {record: call_result}
+    end
+  end
+
+  def call_result_params
+    params.require(:call_results).map do |call_result_params|
+      call_result_params.permit(
+        :id,
+        :user_id,
+        :appointment_id,
+        :cancel_reason,
```

Comment:
`result_type` has three possible values: 

- agreed to visit
- remind to call later
- removed from overdue list

`remove_reason` has 7 possible values right now, and they all indicate reasons why a patient was `removed from overdue list`. Eg: `died`, `invalid phone number` etc.

Its better to model them separately rather than flattening them into a single enum where 7 of 9 values need to be interpreted as `removed from overdue list`

Reply:
Ah very cool very cool. Makes sense to me. Also `remove_reason` :chef_kiss: